metagraph/src/cli/transform_annotation.cpp

```diff
@@ -434,26 +417,45 @@ int transform_annotation(Config *config) {
                 rb_brwt_annotator->serialize(config->outfbase);
                 break;
             }
-            case Config::RowDiff: {
-                logger->trace("Loading graph...");
-                graph::DBGSuccinct graph(2);
-                graph.load(config->infbase);
-                logger->trace("Coverting annotation from column to row-compressed...");
-                RowCompressed<> row_annotator(annotator->num_objects());
-                convert_to_row_annotator(*annotator, &row_annotator, get_num_threads());
-                annotator.reset();
+        }
 
-                logger->trace("Annotation converted in {} sec", timer.elapsed());
-                timer.reset();
-                logger->trace("Converting to row-diff...");
-                std::unique_ptr<annot::RowDiffAnnotator> anno
-                        = convert_to_row_diff(graph, std::move(row_annotator),
-                                              get_num_threads(), config->max_path_length);
-                logger->trace("Annotation converted in {} sec", timer.elapsed());
-                logger->trace("Serializing to '{}'", config->outfbase);
-                anno->serialize(config->outfbase);
-                break;
+    } else if (input_anno_type == Config::RowDiff) {
+        if (config->anno_type != Config::BRWTRowDiff
+            && config->anno_type != Config::ColumnCompressed) {
+            logger->error(
+                    "Only conversion to column compressed and brwt_rowdiff supported for "
+                    "row_diff");
+            exit(1);
+        }
+        if(config->anno_type == Config::BRWTRowDiff) {
+            std::unique_ptr<BRWTRowDiffAnnotator> brwt_annotator;
+            if (config->infbase.empty()) { // load all columns in memory and compute linkage on the fly
+                logger->trace("Loading annotation from disk...");
+                auto annotator = std::make_unique<RowDiffAnnotator>();
+                if (!annotator->merge_load(files))
+                    std::exit(1);
```

Comment:
`merge_load` prints a more informative message, and say exactly which annotation file couldn't be loaded.

Reply:
Not all annotators do it. I saw `annotator`, which doesn't specify the type.
Some annotators may not print error messages inside `load()`, they just return `false`.

So if we rely on certain behavior in this annotator, would be good to call it accordingly, e.g., `row_diff_anno` instead of `annotator`.

But anyway, adding an extra error message won't hurt (but it's ok not to add if you indeed rely on logging inside the method)

Reply:
Renamed to `row_diff_anno`